src/magneticField/turbulentField/PlaneWaveTurbulence.cpp

```diff
@@ -306,15 +283,23 @@ Vector3d PlaneWaveTurbulence::getField(const Vector3d &pos) const {
 		    cos_arg, (_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC));
 
 		// we now compute s, which will be the input parameter to our polynomial
-		// approximation of cos(pi/2*x) between 0 and 1 the andnot_pd is just a
-		// fast way of taking the absolute value
+		// approximation of cos(pi/2*x) between 0 and 1
 		__m256d s = _mm256_sub_pd(cos_arg, q);
 
-		// the following based on the int extraction process described here:
+		// the following is based on the int extraction process described here:
 		// https://stackoverflow.com/questions/41144668/how-to-efficiently-perform-double-int64-conversions-with-sse-avx/41223013
+		//
 		// we assume -2^51 <= q < 2^51 for this, which is unproblematic, as
 		// double precision has decayed far enough at that point that the cosine
 		// would be useless anyway.
+		//
+		// explanation: the mantissa of a double-precision
+		// float has 52 bits (excluding the sign bit). if |q| > 2^51, the msb
+		// of the mantissa has a place value of at least 2^50. (the actual
+		// most significant bit is always 1 and isn't stored.) this means that
+		// the lsb of the mantissa has a place value of at least 2^(-1), or 0.5.
+		// for a cos(pi*x), this corresponds to a quarter of a period (pi/2),
+		// so at this point the cosine isn't really a useful cosine anymore.
```

Comment:
"...the **implemented** cosine isn't a really usefule cosine anymore."
Otherwise I do not really get the last sentence.

Reply:
Hmm, I've just added a little bit more explanation on this, but I'm not sure how well that works. What I was trying to say is that, fundamentally, a double value that is greater than 2⁵¹ can only be accurate to an absolute error of 0.5. (Does this make sense? I'm kind of struggling with how to say this.) So what this means is that for any implementation of cos(pi*x), if x is larger than 2⁵¹, the implementation could still return the correct value for the given input x – it's just that that x is necessarily rounded to 0.5, which means that the function is only sampled 4 times per cycle, and jumps by +/-1 between consecutive samples.

Reply:
I think, now I got it. Thanks for explanation.

Reply:
Great! I'm glad I made sense. Do you think that the updated comment is able to convey this as well, or should I maybe put some more thought into that?